### ISSUE TITLE
[BUG] Passing a value between 0 and 1 to TICKERHANDLER causes KeyError: 'storekey'

### ISSUE THREAD

**Issue by pruuk** [evennia/evennia#2235]
#### Describe the bug
I'm trying to build a 'twitch' style combat system that uses the TICKERHANDLER to control when characters or mobs attack. The simplified system works fine up until you create a ticker with an interval less than 1 second. Example: Attempting to add a new ticker for .5 (I've also tried passing in 0.5) throws an KeyError.

#### To Reproduce
1. tickerhandler.add( <any value between 0 and 1>, <callback>, <etc>)
2. See error
Traceback (most recent call last):
  File "/home/calabedavis/muddev/evennia/evennia/commands/cmdhandler.py", line 644, in _run_command
    ret = cmd.func()
  File "./commands/command_characters.py", line 144, in func
    self.build_tickers(attacker, defender)
  File "./commands/command_characters.py", line 157, in build_tickers
    tickerhandler.add(interval=0.95, callback=attacker.at_attack_tick, idstring=ticker_id, persistent=False)
  File "/home/calabedavis/muddev/evennia/evennia/scripts/tickerhandler.py", line 522, in add
    self.ticker_pool.add(store_key, *args, **kwargs)
  File "/home/calabedavis/muddev/evennia/evennia/scripts/tickerhandler.py", line 254, in add
    log_err(_ERROR_ADD_TICKER.format(store_key=store_key))
KeyError: 'storekey'

#### Expected behavior
Ticker should be created with an interval less than 1 second.

#### Environment, Evennia version, OS etc
Evennia version: 0.9.0 (rev ea6f245f5)
         OS: posix
         Python: 3.8.2
         Twisted: 20.3.0
         Django: 2.2.16

#### Additional context
I can use a workaround of creating multiple tickers for characters/NPCs that attack more than once per second, but I'd like to avoid that as a long term solution because it creates a lot of tickers.

**Comment by Griatch** [comment]
@pruuk Generally, if you create a ticker with a faster tick that 1s in a text game, you are probably doing something wrong in your design. A mob that attacks multiple times per second will spam the user relentlessly for little practical purpose, and updating any data faster than once per second is more than likely useless. The normal way to handle this is to do things on-demand (when the user asks for info, not really relevant for combat) or to aggregate the results to longer time intervals: That is, if you don't display something multiple times per second (which you shouldn't) you can just as well tell the user at the full second that two attacks were made at them and the results of them. The user won't notice the difference - if you really care about sub-second precision when calculating the results, you can still take into account the exact time the user responded with their twitchy response (this kind of gameplay invites client scripting and triggers but I suppose you are planning for that)

So, this is likely an anti-pattern and you should consider changing the design of your combat. Nevertheless, I'll take a look to see what goes wrong in the storage of the ticker.

**Comment by pruuk** [comment]
Thanks for taking a look and for the advice.

I'll see what I can do to redesign. I originally tried setting up my combat system using a script like in the tutorial for turn-based combat, but had better luck getting my system working using tickerhandler. I think this mostly comes down to my inexperience as a programmer. I'm barely past the "I can code my way out of a paper bag" stage.

One additional request (low priority): It'd be nice to be able to kill individual tickers that are tied to a character/NPC by using tickerhandler.remove(<character/mob>).

**Comment by Griatch** [comment]
> One additional request (low priority): It'd be nice to be able to kill individual tickers that are tied to a character/NPC by using tickerhandler.remove(<character/mob>).
You normally do this by specifying the exact ticker you want to remove. You can have any number of tickers on the character/mob, how would the system know which one to kill with this syntax? Or should it kill all on that object, you mean?

**Comment by pruuk** [comment]
It's a good point. I think I'm probably designing around something that doesn't need to be there. But having a way to kill all the tickers on an object for testing might be useful. Probably not as useful after testing.

Another note:
I somehow managed to break my mud last night when I cleared all the tickers (including the tutorial world). I haven't been able to reproduce the problem, but it happened after I tried using tickerhandler.clear(<mobname>) to see what would happen. It cleared all the tickers as expected, but then the mud started throwing errors on any tickerhandler code, including code that was #commented out. If I see the bug again, I'll report it, but I have nothing to give you to work from atm.
